### DISCUSSION TITLE
Trying to make a menu system with Typer

### DISCUSSION THREAD

**Discussion by BruceEckel** [fastapi/typer#719] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
import readchar  # Reads a single character, no CR required

app = typer.Typer(
    no_args_is_help=True,
    add_completion=False,
    context_settings={"help_option_names": ["-h"]},
    rich_markup_mode="rich",
    epilog="Shows up after the help message",
)


@app.command("1")
def one():
    "Command One"
    print("This is Command One!")


@app.command("2")
def two():
    "Command Two"
    print("This is Command Two!")


@app.command("m")
def menu(ctx: typer.Context):
    "Menu Mode"
    print("This is Menu Mode!")


@app.callback()
def main(ctx: typer.Context):
    """
    Attempt to create a menu-driven Typer app
    """
    if ctx.invoked_subcommand == "m":
        print(ctx.get_help())
        while (key := readchar.readchar()) not in "12":
            print(key)
        print(f"You selected {key}")
        match key:
            case "1":
                ctx.call_on_close(one)
            case "2":
                ctx.call_on_close(two)


if __name__ == "__main__":
    app()
```

### Description

I'm trying to create a menu-driven system using Typer. Ideally:
1. If you run the command with no arguments, it prints the help info and waits for a (single-keypress) input.
2. User presses a key to select command, and the program executes the command, then goes back to #1: displays help and waits for input.
3. Program keeps returning to the menu until user explicitly quits.

The example above does work, and you can see I've delved a bit into the `typer.Context`, but I'm wondering if I'm on the right path. It seems like there might be an easier/better approach and I would appreciate any pointers or ideas.

### Operating System

Linux, Windows, macOS

### Operating System Details

Would definitely like this to be a cross-platform solution.

### Typer Version

0.9.0

### Python Version

Python 3.12.0

### Additional Context

Seems like a solution/example that others might benefit from. If this can be figured out perhaps I'll blog about it and if you like that it could be incorporated somewhere into your docs.

**Comment by MaKaNu** [comment]
Looks good so far. What would look more awesome would be that the help show multiple menus and than after selecting one of the menu only the options are shown. I might look deeper into that.
